taskcluster_util/taskcluster_download.py

```diff
@@ -93,19 +93,19 @@ def run(self):
             task_id = self.options.task_id
 
         artifact_downloader = Downloader(connection_options)
-        if self.options.aritfact_name is None and self.options.dest_dir is None:
-            # no artifact_ and dest_dir, then get the latest artifacts list
+        if self.options.aritfact_name is None:
+            # no artifact_name, then get the latest artifacts list
             self.show_latest_artifacts(artifact_downloader, task_id)
-        elif self.options.aritfact_name is None or self.options.dest_dir is None:
-            # if one of artifact_name and dest_dir is specified, return error
-            print('### Please specify the artifact name and dest folder at the same time.\n')
-            exit(-1)
         else:
-            # has artifact_name and dest_dir, then download it
-            abs_dest_dir = os.path.abspath(self.options.dest_dir)
-            if os.path.exists(abs_dest_dir) and (not os.path.isdir(abs_dest_dir)):
-                print('### {} is not a folder.\n'.format(abs_dest_dir))
-                exit(-1)
+            # has artifact_name, then download it
+            if self.options.dest_dir:
```

Comment:
I think we can extract this if/else into a method. So the `run()` method will be smaller :)
``` python

class NotAFolderException(Exception):
    def __init__(self, path):
        super(Exception, self).__init__()
        self.message = '{} is not a folder.'.format(path)

class DownloadRunner(object):
    def _get_output_directory_path(self):
        if self.options.dest_dir:
            abs_dest_dir = os.path.abspath(self.options.dest_dir)
            if os.path.exists(abs_dest_dir) and (not os.path.isdir(abs_dest_dir)):
                raise NotAFolderException(abs_dest_dir)
        else:
            abs_dest_dir = os.getcwd()
        return abs_dest_dir

    def run(self):
        ...
        if self.options.aritfact_name is None:
            # no artifact_name, then get the latest artifacts list
            self.show_latest_artifacts(artifact_downloader, task_id)
        else:
            # has artifact_name, then download it
            abs_dest_dir = self._get_output_directory_path()
            print('### The destination folder is [{}]'.format(abs_dest_dir))
        ...
```

What do you think?